src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -214,13 +206,7 @@ class SourcesTree extends Component<Props, State> {
     if (!nodeHasChildren(item)) {
       const obj = item.contents.get("id");
       const source = sources.get(obj);
-      if (source && source.get("isBlackBoxed")) {
-        return <img className="blackBox" />;
-      }
-
-      const sourceType = sourceTypes[getExtension(source)];
-      const classNames = classnames("source-icon", sourceType || "file");
-      return <img className={classNames} />;
+      return <img className={getSourceClassnames(source.toJS())} />;
```

Comment:
@darkwing I've only extracted the logic that dealt directly with source. It sounded like you were potentially looking for more of a refactor ?

Reply:
I like what you've done :)